### ISSUE TITLE
.hyper_plugins and .hyper.js location when XDG_CONFIG_HOME is not set (or is set but Hyper still searches for wrong location) and potential fix

### ISSUE THREAD

**Issue by maxdevjs** [vercel/hyper#6789]
- [ x] I am on the [latest](https://github.com/vercel/hyper/releases/latest) Hyper.app version
- [ x] I have searched the [issues](https://github.com/vercel/hyper/issues) of this repo and believe that this is not a duplicate

> I found at least https://github.com/vercel/hyper/issues/6616 (for Mac). Nonetheless, I am opening this issue for Linux and as a clarification on the root cause.

- (irrelevant in this case) **OS version and name**: Solus 4.3 Fortitude
- **Hyper.app version**: 3.0.0
- (irrelevant in this case) **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: 
- **Relevant information from devtools** _(CMD+ALT+I on macOS, CTRL+SHIFT+I elsewhere)_: 
- **The issue is reproducible in vanilla Hyper.app**:  with downloaded `Hyper-3.3.0.AppImage`and not "patched" cloned repo

## Issue

Related to https://github.com/vercel/hyper/issues/6616

TLDR: relevant spot is https://github.com/vercel/hyper/blob/ef5419b5200a04b244e9d507baf6f4e5cf5bad9c/app/config/paths.ts#L16

The downloaded `Hyper-3.3.0.AppImage` creates `$HOME/.config/Hyper` (capitalized `Hyper`)

- probably it checks for `$HOME/.config/hyper` (lowercase `h`) as per 
https://github.com/vercel/hyper/blob/ef5419b5200a04b244e9d507baf6f4e5cf5bad9c/app/config/paths.ts#L16
  - it fails to find `$HOME/.config/Hyper` (searches for `$HOME/.config/hyper` (lowercase `h`))
    - then creates both `.hyper_plugins` and `.hyper.js` also if `XDG_CONFIG_HOME` is set

- same behavior found in cloned repo 
  - until when https://github.com/vercel/hyper/blob/ef5419b5200a04b244e9d507baf6f4e5cf5bad9c/app/config/paths.ts#L16 is updated to `? join(process.env.XDG_CONFIG_HOME, 'Hyper')` (capitalized `Hyper`)

At this point both `.hyper_plugins` and `.hyper.js` are correctly created in  `XDG_CONFIG_HOME/Hyper`, correct configuration and plugins are loaded.

I pushed the fix to my [fork](https://github.com/maxdevjs/hyper/blob/022f8b55913700f296d4caff245d57a3c9808579/app/config/paths.ts#L16). If all this is correct (both Linux and Mac), I'd like to send a PR.
